mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1047,6 +1076,12 @@ private[ml] object RandomForest extends Logging {
       numNodesInGroup += 1
       memUsage += nodeMemUsage
     }
+    if (memUsage > maxMemoryUsage && maxMemoryUsage != 0) {
```

Comment:
I'm not sure why the check `&& maxMemoryUsage != 0` is needed. The way it is here, you can set `maxMemoryInMB` to 0, and it will train the RF one node at a time and issue no warnings. Then again, I'm not sure why you'd be allowed to set `maxMemoryInMB` to 0 at all.